raiden/storage/sqlite.py

```diff
@@ -1376,10 +1376,20 @@ def get_light_client_messages(self, from_message, light_client):
         cursor = self.conn.cursor()
         cursor.execute(
             """
-            SELECT identifier, message_order, unsigned_message, signed_message, light_client_payment_id, internal_msg_identifier, message_type
+            SELECT identifier,
+                   message_order,
+                   unsigned_message,
+                   signed_message,
+                   light_client_payment_id,
+                   internal_msg_identifier,
+                   message_type
             FROM light_client_protocol_message
             WHERE internal_msg_identifier >= ?
             AND light_client_address = ?
+            AND (
+                message_type NOT IN ('SettlementRequired', 'RequestRegisterSecret', 'UnlockLightRequest')
```

Comment:
question here, what about messages that:
1. have a type ∉ (`SettlementRequired`, `RequestRegisterSecret`, `UnlockLightRequest`), but **also**:
2. have `signed_message IS NOT NULL`

do you want to retrieve these?

Reply:
Well according to the crew, seems that those messages are being retrieved too since the first condition of the OR should be TRUE and that should work.

Reply:
Yes you want.

That scenarios are, for example, when you create a payment from a light client.

The sdk invokes the create payment endpoint and creates an unsigned lockedtransfer. 

Then the sdk gets that msg from the polling and sign it.